Userland/Libraries/LibGfx/PortableFormatWriter.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023, Lucas Chollet <lucas.chollet@free.fr>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/ByteBuffer.h>
+#include <LibGfx/Bitmap.h>
+
+namespace Gfx {
+
+class PortableFormatWriter {
+public:
+    enum class Format {
+        Ascii,
+        Raw,
+    };
+
+    static ErrorOr<ByteBuffer> encode(Gfx::Bitmap const&, Format = Format::Raw);
```

Comment:
Instead of an arg, what do you think about using an Options object like BMPWriter does? Using the named aggregate initializer, that makes for more readable call sites, keeps the Writers consistent, and scales better to multiple options.